### DISCUSSION TITLE
[CH4] 메모리 제어

### DISCUSSION THREAD

**Discussion by innerstella** [innerstella/ES2024#5] [Q&A] [answered]
자바스크립트는 개발자의 직접적인 메모리 제어를 허용하지 않는데 이 이유와 설명

**Comment by Lv1GoM** [comment, marked as answer]
자바스크립트는 개발자의 직접적인 메모리 제어를 허용하지 않음
=> 치명적인 오류를 발생시킬 가능성이 높은 일이기 때문
어떤 치명적인 오류?
메모리 오버플로우, 메모리 누수와 같은 치명적인 버그를 방지
가비지 컬렉션을 이용한 메모리 관리 => 누수가 일어나지 않도록.
가비지 컬렉션은 더 이상 필요하지 않은 객체나 변수를 식별하고 자동으로 메모리에서 제거

```
function createClosure() {
  const data = 'Some data';
  return function() {
    console.log(data);
  };
}

const leakyClosure = createClosure();
```
이 코드에서 leakyClosure는 외부 함수의 data 변수에 대한 참조를 계속 유지하고 있어 해당 함수 스코프의 데이터에 접근 가능. 이로 인해 해당 데이터는 가비지 컬렉션되지 않아 메모리 누수가 발생
